### ISSUE TITLE
Create new project is extremely slow

### ISSUE THREAD

**Issue by stanleyxu2005** [vuejs/vue-cli#1444]
### Version
3.0.0-beta.15

### Reproduction link
[http://vuejs.org](http://vuejs.org)

### Steps to reproduce
vue create new_project

### What is expected?
Install dependencies and then show me a "Done"

### What is actually happening?
Blocked at "Creating project in ...", and it is never finish

**Comment by ITPoker** [comment]
Now things seem to be getting better. I use command : vue create new_project. Faster than a few hours ago

**Comment by CheckMyselfEveryday** [comment]
same problem

**Comment by yyx990803** [comment]
Please provide a bit more information. You have gone through the preset selection process? And it's stuck at exactly at  `Creating project in ...`, without the Git initialization info, and without any yarn/npm install output?

For anyone else commenting in this thread, make sure you are actually having the **exact same** issue, slow installation is unrelated and more likely your npm registry connection problem.

**Comment by stanleyxu2005** [comment]
Dear Evan @yyx990803 

Here are some more details: 
* Physical Location: Shanghai
* OS: Windows 7
* Node: 10.2.1 with built in npm,  and cnpm 5.6.0 + yarn 1.7

The problem is there is no chance for me to choose alternative npm registry, there is also no switcher to show verbose information. 

**Bad one**
I made screenshot of the process. (It never succeed on my machine) I guess, it might download everything overseas with a very slow connection. 
![image](https://user-images.githubusercontent.com/5162687/40893073-03604038-67d2-11e8-9e0e-c1793461f061.png)
![image](https://user-images.githubusercontent.com/5162687/40893076-09e6d4da-67d2-11e8-9288-a005b3a47451.png)
![image](https://user-images.githubusercontent.com/5162687/40893082-1220b440-67d2-11e8-8b18-e18ab8874375.png)

**Good one**
My current workaround is to create project with switch `-d`, it looks much faster.
![image](https://user-images.githubusercontent.com/5162687/40893151-ef35968e-67d2-11e8-9b35-6d3a9126c739.png)

**Comment by ziyi2** [comment]
same problem  Node: v10.3.0  Npm: 6.1.0       OS: WIN10

**Comment by ys930126** [comment]
Emmmm , Has Same Problem  When I  vue create project , Solute It By Use '-d'  Commad

**Comment by Bouke** [comment]
I seem to be running into the same problem:

```powershell
[...] [feature/vue ≡ +3 ~0 -0 !]> .\node_modules\.bin\vue create
my-project-name


Vue CLI v3.0.0-beta.15
? Please pick a preset: Manually select features
? Check the features needed for your project: Babel, TS, Linter
? Use class-style component syntax? Yes
? Use Babel alongside TypeScript for auto-detected polyfills? Yes
? Pick a linter / formatter config: TSLint
? Pick additional lint features: Lint on save
? Where do you prefer placing config for Babel, PostCSS, ESLint, etc.? In dedicated config files
? Save this as a preset for future projects? No
? Pick the package manager to use when installing dependencies: Yarn


Vue CLI v3.0.0-beta.15
-  Creating project in [...]\my-project-name.
```

This is on Win10, yarn 1.7.1, node 10, from powershell. According to PowerShell's window title the hanging process is "npm".

**Comment by yyx990803** [comment]
The original issue cannot be reproduced, so I am closing this.

If you are just experiencing slow dependency installation (i.e. you see the install bar), make sure you are using a faster registry for your area (e.g. Taobao mirror if you are in China).

**Comment by lipeunim** [comment]
> The description is a bit vague, please provide a bit more information. You have gone through the preset selection process? And it's stuck at exactly at `Creating project in ...`, without the Git initialization info, and without any yarn/npm install output?
> 
> What OS are you on, what are your Node/npm/yarn versions? Can you provide a screenshot, or better, a video or exactly what happens?
> 
> For anyone else commenting in this thread, make sure you are actually having the **exact same** issue, slow installation is unrelated and more likely your npm registry connection problem.

![screen shot 2019-01-16 at 12 18 33 pm](https://user-images.githubusercontent.com/37588172/51245768-f220e700-1988-11e9-8039-1fef7f1de4b6.JPG)

**Comment by iqmal852** [comment]
Hi there, im still having this issue

**Comment by javadmokhtari** [comment]
still having this issue. it takes about 10 minutes to create a project even with high speed connection.
command: vue create project_name

**Comment by Akryum** [comment]
@javadmokhtari Do you have an Anti-virus software running?

**Comment by javadmokhtari** [comment]
@Akryum no, I don't have antivirus software. And I have Linux Mint OS (maybe useful for you).

**Comment by laxsmi** [comment]
>  And I have Linux Mint OS (maybe useful for you).

I am experiencing the same issue with Linux mint.

**Comment by iqmal852** [comment]
I fixed it by disable taobao registry

**Comment by namnguyen2091** [comment]
@iqmal852 where do I can do this?

**Comment by namnguyen2091** [comment]
> @Akryum no, I don't have antivirus software. And I have Linux Mint OS (maybe useful for you).

the same with me, it's still running after 2 hours.

**Comment by Akryum** [comment]
@namnguyen2091 https://cli.vuejs.org/config/#global-cli-config

**Comment by namnguyen2091** [comment]
> @namnguyen2091 https://cli.vuejs.org/config/#global-cli-config

thanks so much, it's much faster:D

**Comment by Baronsindo** [comment]
2019 Still Slow can't it be local then get updated when it needs to ?

**Comment by bala41288** [comment]
I had a similar issue both for creating new projects as well as plugin installation. After digging so much for several hours, I found out that it was my network proxy settings.

**Comment by CyrilHskt** [comment]
> @namnguyen2091 https://cli.vuejs.org/config/#global-cli-config

Thanks !

**Comment by freelancer254** [comment]
Disable your antivirus and clear some resource intensive processes.
It should take about 10 minutes.
Also check: vue config
Resolved path: ...\.vuerc
 {
  "useTaobaoRegistry": false
}

**Comment by Nartek01** [comment]
Im using ubuntu 18.04 LTS and is having the same issue that after I've downloaded @vue/cli when I ran vue create test or vue create new_project or vue create -d new_project it would seems like the download is very slow, but is not stuck since I can see that the download bar is switchting to different files. I've managed to fix this problem by updating node.js below are the commands that I ran on the terminal which fixed the issue.

sudo apt-get update 
sudo apt-get install build-essential checkinstall libssl-dev 

curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.32.1/install.sh | bash 
sudo apt install npm

*exit the terminal and start it again*

sudo npm cache clean -f
sudo npm install -g n
sudo n stable


Maybe some of the commands was unecessary, but this was exactly what I wrote to fix the issue that made vue cli seems to be downloading very slow or not at all to be able to download vue cli and it's dependencies under 3-4 minutes with 100mbit connection.
